src/System.Net.Security/src/System/Net/Unix/CertModule.cs

```diff
@@ -0,0 +1,75 @@
+// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+using Microsoft.Win32.SafeHandles;
+using System.Net.Security;
+using System.Runtime.InteropServices;
+using System.Security.Cryptography.X509Certificates;
+
+namespace System.Net
+{   
+    internal partial class CertModule : CertInterface
+    {
+        #region internal Methods
+        internal override SslPolicyErrors VerifyRemoteCertName(X509Chain chain, bool isServer, string hostName)
+        {
+            SslPolicyErrors sslPolicyErrors = SslPolicyErrors.None;         
+
+            return sslPolicyErrors;
+        }
+
+        //
+        // Extracts a remote certificate upon request.
+        //
+        internal override X509Certificate2 GetRemoteCertificate(SafeDeleteContext securityContext, out X509Certificate2Collection remoteCertificateStore)
+        {
+            remoteCertificateStore = null;
+
+            if (securityContext == null)
+            {
+                return null;
+            }
+
+            GlobalLog.Enter("SecureChannel#" + Logging.HashString(this) + "::RemoteCertificate{get;}");
+            X509Certificate2 result = null;
+            SafeFreeCertContext remoteContext = null;
+            try
+            {
+                int errorCode = SSPIWrapper.QueryContextRemoteCertificate(GlobalSSPI.SSPISecureChannel, securityContext, out remoteContext);
+                if (remoteContext != null && !remoteContext.IsInvalid)
+                {
+                    result = new X509Certificate2(remoteContext.DangerousGetHandle());
```

Comment:
There doesn't appear to be anything keeping remoteContext alive after the call to get its DangerousGetHandle, which means it's theoretically possible it could get collected and its handle released boefre X509Certificate2's ctor gets a chance to use and dup it.  This should really be surroundd by DangerousAddRef/DangerousRelease calls, but if not that GC.KeepAlive should at least be used.